File: parsers/dotnet/Test.Taml.NET/BaseFixture.cs
using System.IO;

namespace Test.Taml.NET
{
	public abstract class BaseFixture
	{
		protected StreamReader Sample { get; }

		protected abstract string SampleFilename { get; }

		public BaseFixture() 
		{
			Sample = new StreamReader(File.OpenRead(SampleFilename));
		}

	}
}

File: src/JVM/taml4j-parser/src/test/java/com/taml/BaseFixture.java
package com.taml;

import java.io.*;
import java.net.URL;

/**
 * @author Luca Taddeo - Lucalas
 */
public abstract class BaseFixture {
	protected BufferedReader getReader() {
		URL url = Thread.currentThread().getContextClassLoader().getResource(getSampleFilename());
		try {
			return new BufferedReader(new InputStreamReader(url.openStream()));
		} catch (IOException e) {
			e.printStackTrace();
		}
		return null;
	}

	protected abstract String getSampleFilename();
}
